tekton/release-pipeline.yaml

```diff
@@ -85,6 +85,8 @@ spec:
       params:
         - name: package
           value: $(params.package)
+        - name: flags
+          value: -ldflags "-X github.com/tektoncd/pipeline/pkg/version.PipelineVersion=$(git describe --abbrev=0)"
```

Comment:
I think this won't work with the order we do things today, i.e.:
- lint, unit tests, build
- publish images, create release yaml, push it to the bucket
- test the release (well, this is WIP)
- create the release in github, push the tag

The idea is that we would like to test the release E2E before we push the tag.
We could also create the tag first, and in case the release is broken:
- if the release is broken because of an issue in the release script, fix the issue and repeat the process
- if the release is broken because of a bug that was missed in CI, fix the issue and release a new minor version

Reply:
@afrittoli  Thanks for checking this out. I have updated this with the `versionTag` param and have also added an override in `.ko.yaml` and `.ko.release.yaml` cc @vdemeester 

Do check it out